internal/controller/streams.go

```diff
@@ -236,6 +272,26 @@ func connectAndStreamCiliumNetworkFlows(logger *zap.SugaredLogger, sm *streamMan
 	return nil
 }
 
+// connectAndStreamFalcoNetworkFlows creates networkFlowsStream client and begins the streaming of network flows.
+func connectAndStreamFalcoNetworkFlows(logger *zap.SugaredLogger, sm *streamManager) error {
+	falcoCtx, falcoCancel := context.WithCancel(context.Background())
+	defer falcoCancel()
+	sendFalcoNetworkFlows, err := sm.streamClient.client.SendKubernetesNetworkFlows(falcoCtx)
+	if err != nil {
+		logger.Errorw("Failed to connect to server", "error", err)
+		return err
+	}
+	sm.streamClient.networkFlowsStream = sendFalcoNetworkFlows
```

Comment:
More generally, I still don't see the point of defining and setting this `networkFlowsStream` field, as the stream is only used by the one call to `StreamFalcoNetworkFlows` below. So it is not really part of the state. It's only used in the context of this one execution stack.
This should rather be passed as a parameter to `StreamFalcoNetworkFlows` (resp. `StreamCiliumNetworkFlows`).